### ISSUE TITLE
[BUG] Milestones Table View sort by Date ignores year

### ISSUE THREAD

**Issue by ashelman-ec** [Leantime/leantime#918]
**Describe the bug**
If a project has multiple milestones, whose dates cross a year boundary, when in Milestones Table View, there is no way to sort the milestones in chronological order. Clicking on the **Planned Start Date** or **Planned End Date** seems to account for the month & day only, but ignores the year. Thus the milestones are not in order.
_(Note: milestones do display correctly in the Gantt View.)_

**To Reproduce**
1. Create a milestone that begins in December of this year, and a second that begins in January of next year.
2. Go to 'Milestones'
3. Click on 'Table View'
4. Click on 'Planned Start Date'. Column title turns dark, and Up arrow displays.
5. See error (January milestone is first).

For an example with more milestones, see screenshot below.

**Expected behavior**
Year should be taken into account, so the expected sort order is as written in red in screenshot below.
In other words, instead of this order:
01/01/2023, ..., 04/01/2023, 04/21/2022, ..., 12/01/2022
we should have this order:
2022-04-21, ... 2022-12-01, 2023-01-01, ..., 2023-04-01

**Leantime Version**
2.1.9

**Server**
Apache

**PHP / MySQL Version**
PHP 7.4
Mysql 8.0

**Additional context**
I realize my Leantime version (v2.1.9) is not the latest (v2.2.7). I don't see this bug fix mentioned in the release notes, but if it has already been fixed let me know & I will upgrade.

![TableView-sort-up](https://user-images.githubusercontent.com/104984057/186666758-c0ddd5fb-8c78-4dd9-a30f-c69af2197acd.jpg)

**Comment by marcelfolaron** [comment]
Thanks for submitting. You are right, this is still an issue in the latest
version.

Ill get a fix out.

**Comment by ashelman-ec** [comment]
Thanks! I'm enjoying using LeanTime for managing my project.

Note: Implementing this will probably also help in implementing #410.
